hackathon_site/event/api_views.py

```diff
@@ -66,3 +66,38 @@ def post(self, request, *args, **kwargs):
         response_serializer = TeamSerializer(profile.team)
         response_data = response_serializer.data
         return Response(data=response_data, status=status.HTTP_201_CREATED,)
+
+
+class JoinTeamView(generics.GenericAPIView, mixins.RetrieveModelMixin):
+    permission_classes = [UserHasProfile]
+    serializer_class = TeamSerializer
+    lookup_field = "team_code"
+    queryset = EventTeam.objects.all()
+
+    @transaction.atomic
+    def post(self, request, team_code, *args, **kwargs):
```

Comment:
Can actually remove `team_code` from here since it's unused (consumed implicitly by `get_object()` now)